lib/matplotlib/axis.py

```diff
@@ -2380,8 +2380,13 @@ def _update_offset_text_position(self, bboxes, bboxes2):
         Update the offset_text position based on the sequence of bounding
         boxes of all the ticklabels
         """
-        x, y = self.offsetText.get_position()
-        top = self.axes.bbox.ymax
+        x, _ = self.offsetText.get_position()
+        if 'outline' in self.axes.spines:
+            # Special case for colorbars:
+            bbox = self.axes.spines['outline'].get_window_extent()
+        else:
+            bbox = self.axes.bbox
```

Comment:
Would it make sense to update the `axes.bbox` in Colorbar to be the full spines outline? Move it up a level, so everything is referencing the same full bbox rather than just the text here. I'm not sure if that would mess up the tickers to draw in the extends region or not then...

Reply:
I try to paraphrase: the colorbar axes has additional decorations which are outside the bbox but need to be taken into accout for shifting the offset text? In that case, do we need a concept for "bbox with decorations"?

Reply:
We do, it's called get_tightbbox.  However that is defined on the axes and takes into account the axis, and this exponent is part of the axis, so you get an infinite recursion.  I guess we could change get_tightbbox to accept a list of artists to exclude, but I'm not convinced that is any better than this solution. And again get_tightbbox can be very expensive.

Reply:
Ok. I don't claim to I know how the correct or practical solution. I was just trying to understand what's going on.